src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexFacetedReadOperation.cs

```diff
@@ -39,29 +40,147 @@ public sealed class CoraxIndexFacetedReadOperation : IndexFacetReadOperationBase
+                    facetsByName[result.Key] = facetValues = new Dictionary<string, FacetValues>();
+
+                var metadata = GetFieldMetadata(result.Key);
+
+                var provider = _indexSearcher.TextualAggregation(metadata, forward: result.Value.Options.TermSortMode is not FacetTermSortMode.ValueDesc);
+                using var _ = provider.AggregateByTerms(out result.Value.SortedIds, out var counts);
+
+                var idX = 0;
+                foreach (var term in CollectionsMarshal.AsSpan(result.Value.SortedIds))
+                {
+                    ref var collectionOfFacetValues = ref CollectionsMarshal.GetValueRefOrAddDefault(facetValues, term, out var exists);
+                    if (exists == false)
+                    {
+                        var range = FacetedQueryHelper.GetRangeName(result.Value.AggregateBy, term);
+                        collectionOfFacetValues = new FacetValues(facetQuery.Legacy);
+                        if (needToApplyAggregation == false)
+                            collectionOfFacetValues.AddDefault(range);
+                        else
+                        {
+                            foreach (var aggregation in result.Value.Aggregations)
+                                collectionOfFacetValues.Add(aggregation.Key, range);
+                        }
+                    }
+
+                    collectionOfFacetValues.IncrementCount((int)counts[idX++]);
+                    if (needToApplyAggregation)
+                    {
+                        //not supported yet.
+                        //todo:   ApplyAggregation(result.Value.Aggregations, collectionOfFacetValues, ref reader);
+                    }
+
+                    continue;
+                }
+            }
+
+
+            if (facetsByRange.TryGetValue(result.Key, out facetValues) == false)
+            {
+                facetValues = new();
+                facetsByRange.Add(result.Key, facetValues);
+            }
+
+            var ranges = result.Value.Ranges;
+            foreach (var parsedRange in ranges ?? Enumerable.Empty<FacetedQueryParser.ParsedRange>())
+            {
+                if (parsedRange is not FacetedQueryParser.CoraxParsedRange range)
+                    continue;
+
+                ref var collectionOfFacetValues = ref CollectionsMarshal.GetValueRefOrAddDefault(facetValues, parsedRange.RangeText, out var exists);
+                if (exists == false)
+                    collectionOfFacetValues = new FacetValues(facetQuery.Legacy);
+
+                var fieldMetadata = GetFieldMetadata(range.Field);
+                var aggregationProvider = range.GetAggregation(_indexSearcher, fieldMetadata, true);
+                var count = aggregationProvider.AggregateByRange();
+                collectionOfFacetValues.IncrementCount((int)count);
+                if (needToApplyAggregation)
+                {
+                    //not supported yet.
```

Comment:
?

Reply:
Well, I've disabled using indexed-backed facet queries when this condition is true, because we need to scan like in previous version too. If this is a thing, we just scan like before.

Reply:
What does the code have to look then? Is this a potential true value that can come?